scheduler/core/components/selector/selector.py

```diff
@@ -65,32 +63,34 @@ class Selector(SchedulerComponent):
 
     def __post_init__(self):
         # We need to copy or changing these values will modify the external dictionaries passed in.
-        self.cc_per_site = copy(self.cc_per_site)
-        self.iq_per_site = copy(self.iq_per_site)
+        self.variant_per_site = copy(self.variant_per_site)
 
         if (self.num_nights_to_schedule < 0 or
                 self.num_nights_to_schedule > self.collector.num_nights_calculated):
             raise ValueError(f'Scheduling requested for {self.num_nights_to_schedule} nights, but visibility '
                              f'calculations only performed for {self.collector.num_nights_calculated}. '
                              'Cannot proceed.')
 
-        for site in self.collector.sites - self.cc_per_site.keys():
+        for site in self.collector.sites - self.variant_per_site.keys():
+            logger.warning(f'Selector has no Weather Variant data for {site.name}: setting to {Selector._default_cc.name}')
+            self.variant_per_site[site] = Variant(cc=Selector._default_cc,
+                                                  iq=Selector._default_iq,
+                                                  wind_spd=Selector._wind_spd_bound,
+                                                  wind_dir=Selector._wind_sep)
+
+        for site in self.collector.sites - self.variant_per_site.keys():
```

Comment:
This should all be handled by my PR now, I think... it's `VariantSnapshot` and not `Variant` again, though.